analysers/trials/numberOfProcessInstances.py

```diff
@@ -23,10 +24,95 @@
     .set("spark.cassandra.connection.host", cassandraHost)
 sc = CassandraSparkContext(conf=conf)
 
-data = sc.cassandraTable(cassandraKeyspace, srcTable)\
-        .select("duration") \
+dataRDD = sc.cassandraTable(cassandraKeyspace, srcTable)\
+        .select("process_definition_id", "source_process_instance_id", "end_time", "start_time", "duration") \
         .where("trial_id=? AND experiment_id=?", trialID, experimentID) \
-        .map(lambda r: 1) \
+        .filter(lambda r: r["process_definition_id"] is not None) \
+        .cache()
+
+processes = dataRDD.map(lambda r: r["process_definition_id"]) \
+        .distinct() \
+        .collect()
+
+maxTime = None
+maxID = None
+for p in processes:
+    time = dataRDD.filter(lambda r: r["process_definition_id"] == p) \
+        .map(lambda r: (r["start_time"], r["source_process_instance_id"])) \
+        .sortByKey(1, 1) \
+        .collect()
+    time = time[nToIgnore-1]
```

Comment:
You cannot be sure that you actually have enough process to always find the `nToIgnore-1` one. It should be safe about this and just skip the `p` if you don't find enough processes. Then if the `maxID` will end up to not be set, you just need to use all the data you have.